sky/backends/cloud_vm_ray_backend.py

```diff
@@ -2803,6 +2854,49 @@ def _provision(
             if dryrun:
                 record = global_user_state.get_cluster_from_name(cluster_name)
                 return record['handle'] if record is not None else None
+
+            # We pass handle in the config dict for the new provisioner
+            if 'provision_metadata' in config_dict:
+                # New provisioner is used here.
+                handle = config_dict['handle']
+                provision_metadata = config_dict['provision_metadata']
+                resources_vars = config_dict['resources_vars']
+
+                # setup SkyPilot runtime after the cluster is provisioned
+                cluster_metadata = provision_utils.post_provision_runtime_setup(
+                    repr(handle.launched_resources.cloud),
+                    provision_utils.ClusterName(handle.cluster_name,
+                                                handle.cluster_name_on_cloud),
+                    handle.cluster_yaml,
+                    local_wheel_path=local_wheel_path,
+                    wheel_hash=wheel_hash,
+                    provision_metadata=provision_metadata,
+                    custom_resource=resources_vars.get('custom_resources'),
+                    log_dir=self.log_dir)
+                internal_ips, external_ips = zip(*cluster_metadata.ip_tuples())
+                handle.update_cluster_ips(max_attempts=_FETCH_IP_MAX_ATTEMPTS,
+                                          internal_ips=list(internal_ips),
+                                          external_ips=list(external_ips))
+                handle.update_ssh_ports(max_attempts=_FETCH_IP_MAX_ATTEMPTS)
+
+                # update launched resources
+                handle.launched_resources = handle.launched_resources.copy(
+                    region=provision_metadata.region,
+                    zone=provision_metadata.zone)
```

Comment:
In `ProvisionMetadata`, we should add a comment that the `zone` field should be a single zone.